Jenkinsfile

```diff
@@ -181,28 +184,84 @@ pipeline {
           }
         }
         stage('Release') {
-          options { skipDefaultCheckout() }
+          options {
+            skipDefaultCheckout()
+            timeout(time: 12, unit: 'HOURS')
+          }
+          environment {
+            HOME = "${env.WORKSPACE}"
+          }
           when {
             beforeAgent true
-            tag pattern: '@elastic/apm-rum@\\d+\\.\\d+\\.\\d+$', comparator: 'REGEXP'
+            beforeInput true
+            allOf {
+              branch 'master'
+              expression { return params.release }
+            }
           }
           stages {
-            stage('Opbeans') {
-              environment {
-                REPO_NAME = "${OPBEANS_REPO}"
+            stage('Notify') {
+              options { skipDefaultCheckout() }
+              steps {
+                deleteDir()
+                unstash 'source'
+                unstash 'cache'
+                dir("${BASE_DIR}") {
+                  prepareRelease() {
+                    script {
+                      sh(label: 'Lerna version dry-run', script: 'lerna version --no-push --yes', returnStdout: true)
+                      def releaseVersions = sh(label: 'Gather versions from last commit', script: 'git log -1 --format="%b"', returnStdout: true)
+                      log(level: 'INFO', text: "Versions: ${releaseVersions}")
+                      emailext subject: "[${env.REPO}] Release ready to be pushed", to: "${NOTIFY_TO}",
+                               body: "Please go to ${env.BUILD_URL}input to approve or reject within 12 hours.\n Changes: ${releaseVersions}"
+                      input(message: 'Should we release a new version?', ok: 'Yes, we should.',
```

Comment:
Input approval with a text-field that contains the body for the last commit message, in other words, it will show the list of packages and versions that will be released.

The text-field is not read-only at all, but only informative, even you can edit the text it will do nothing. Unfortunately, I did not find another way to show easily what are the changes that will be applied. It might mislead if you see a text-field in the release, but that's the reason I added a message with `Look at the versions to be released. They cannot be edited here`

What do you think about this approach?

Reply:
This looks really good as we do the exact similar thing on local as well.